src/core/api/index.js

```diff
@@ -62,27 +63,38 @@ export function callApi({
 
   return fetch(apiURL, options)
     .then((response) => {
-      return response.clone().json()
-        .then(
-          (jsonResponse) => ({ response, jsonResponse }),
-          (error) => {
-            log.warn('Could not parse response as JSON:', error);
-            return response.clone().text().then((textResponse) =>
-              ({ response, jsonResponse: { text: textResponse } })
-            );
-          }
-        );
+      function makeTextResponseJson(textResponse) {
+        return { jsonResponse: { text: textResponse }, response };
+      }
+
+      const contentType = response.headers.get('Content-Type').toLowerCase();
+
+      // This is a bit paranoid, but we ensure the API returns a JSON response
+      // (see https://github.com/mozilla/addons-frontend/issues/1701).
+      // If not we'll store the text response in JSON and log an error.
+      // If the JSON parsing fails; we log the error and return an "unknown
+      // error".
+      if (contentType === 'application/json') {
+        return response.json()
+          .then((jsonResponse) => {
+            return { response, jsonResponse };
+          });
```

Comment:
I guess it's fine for now though, I'll move it back to a one-liner, especially as this is now cleaner.